Prevent brute force login attempts using `django-axes`

yenaing-oo opened the pull request:
Closes #844

danloveg commented on app/app/settings/base.py:
```diff
@@ -43,8 +51,18 @@
 ]
 
+# Axes configuration
+AXES_FAILURE_LIMIT = config("AXES_FAILURE_LIMIT", default=5, cast=int)
+AXES_COOLOFF_TIME = config("AXES_COOL_OFF_TIME", default=0.01, cast=float)  # in hours
```
Can you add these to the documentation? The documentation doesn't have to be extensive, but maybe you could link to the django-axes documentation for each of these

danloveg commented on app/app/settings/base.py:
```diff
@@ -43,8 +51,18 @@
 
+# Axes configuration
+AXES_FAILURE_LIMIT = config("AXES_FAILURE_LIMIT", default=5, cast=int)
+AXES_COOLOFF_TIME = config("AXES_COOL_OFF_TIME", default=0.01, cast=float)  # in hours
```
I think thirty six seconds is too short. Let's make it half an hour by default (people can always change it)

```suggestion
AXES_COOLOFF_TIME = config("AXES_COOL_OFF_TIME", default=0.5, cast=float)  # in hours
```

yenaing-oo replied:
Yes, 36 seconds is way too short. I initially had it at 15 mins but I must've accidentally changed it during testing.

Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/856/commits/94d8aa86bfc39e7278bc2f35ad0838df8734eb62

danloveg reviewed (changes requested):
Can you add a warning message if the user has two or fewer remaining attempts to log in before they are locked out?

yenaing-oo commented:
> Can you add a warning message if the user has two or fewer remaining attempts to log in before they are locked out?

That's a good idea, I've added that here: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/856/commits/ecfa63c67456aba1d3732fa1bc829a4010c31a5e

I've also added verification for those settings, as well as tests for the toast events.

yenaing-oo commented on app/recordtransfer/views/account.py:
```diff
@@ -122,18 +129,56 @@ def form_valid(self, form: AuthenticationForm) -> HttpResponse:
+                    )
+                    % num_tries_left,
+                },
+            )
```
That's a pretty neat thing to have - will be using this in the future.

danloveg reviewed (approved):
I will merge this in for August 2025
